core/state-machine/src/proving_backend.rs

```diff
@@ -29,6 +30,47 @@ use crate::trie_backend::TrieBackend;
 use crate::trie_backend_essence::{Ephemeral, TrieBackendEssence, TrieBackendStorage};
 use crate::{Error, ExecutionError, Backend};
 
+/// A proof that some set of key-value pairs are included in the storage trie.
```

Comment:
Looks pretty reasonable. Where is this verification process (walking the nodes and building the sub-tree) that you mentioned in the docs happening?

Reply:
It's a bit spread around, but mostly `create_proof_check_backend`.